### PR TITLE
Optimized metadata handling

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

**Comment by hbs** [comment]
PTAL

File: warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java [hunk 1]
```diff
@@ -8754,7 +8781,17 @@ public static GTSEncoder shrinkTo(GTSEncoder encoder, int newsize) throws IOExce
...
+    
+    //
+    // If chunkcount is Integer.MAX_VALUE, we do not care about the number of chunks,
+    // so if keepempty is false we only need to iterate over the chunks from newestChunk to
+    // oldestChunk
+    //
+    
+    if (!keepempty && zeroChunkCount) {
+      lastchunkid = newestChunk;
+    }
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java:8793]: You must add `if (lastchunkid > newestChunk)`.

Also, the same idea as above comment can apply here: drop `zeroChunkCount` and update `chunkcount` with `chunkcount -= (lastchunkid - newestChunk) / chunkwidth`

**Review by ftence — CHANGES_REQUESTED** [review]

**Comment by hbs** [comment]
PTAL

File: warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java [hunk 2]
```diff
@@ -8417,6 +8417,24 @@ public static GTSEncoder shrinkTo(GTSEncoder encoder, int newsize) throws IOExce
       }            
     }
 
+    //
+    // If chunkcount is 0 adjust lastchunk so it contains the last tick of the GTS
+    // or the overlap
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java:8422]: Please update the comment to reflect the code below.

File: warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java [hunk 3]
```diff
@@ -8751,10 +8783,27 @@ public static GTSEncoder shrinkTo(GTSEncoder encoder, int newsize) throws IOExce
...
     if (0 == lastchunk) {
       lastchunkid = newestChunk;
     }
-
+    
+    //
+    // If chunkcount is Integer.MAX_VALUE, we do not care about the number of chunks,
+    // so if keepempty is false we only need to iterate over the chunks from newestChunk to
+    // oldestChunk.
+    // If chunkcount is ot 0, adjust the number of chunks
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java:8795]: typo "ot" -> "not".

**Review by ftence — CHANGES_REQUESTED** [review]

**Comment by hbs** [comment]
PTYAL

**Review by ftence — APPROVED** [review]

**Pull request merged**